network/wsPeer_test.go

```diff
@@ -37,3 +38,29 @@ func TestCheckSlowWritingPeer(t *testing.T) {
 	require.Equal(t, peer.CheckSlowWritingPeer(now), true)
 
 }
+
+// TestGetNonce tests if the values are incremented correctly
+func TestGetNonce(t *testing.T) {
+	peer := wsPeer{}
+	doneChannel := make(chan bool, 1)
+	for x := 0; x < 200; x++ {
+		go func() {
+			ans := peer.getNonce()
+			val, _ := binary.Uvarint(ans)
+			if val == 200 {
+				doneChannel <- true
+			}
+		}()
+	}
+	maxWait := time.After(2 * time.Second)
+	done := false
+	select {
+	case <-doneChannel:
+		done = true
+	case <-maxWait:
```

Comment:
I'm not sure if this test is the best way to verify the getNonce function.

Reply:
Rewrote the test to catch the bug in the getNonce() implementation.